docs/assignment.md

```diff
@@ -1,3 +1,5 @@
+_Note that this document describes a process that is now deprecated. For more information, see documentation on [adding a new language](https://github.com/github/semantic/blob/master/docs/adding-new-languages.md)._
```

Comment:
Later once we've dropped support for a la carte ASTs entirely and removed `Data.Syntax`, we can delete this file altogether or move it to an archive.